hypershift-operator/controllers/hostedcluster/internal/platform/aws/aws.go

```diff
@@ -230,93 +234,46 @@ func (p AWS) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, hcp *hy
 func (p AWS) ReconcileCredentials(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN,
 	hcluster *hyperv1.HostedCluster,
 	controlPlaneNamespace string) error {
-	// Reconcile the platform provider cloud controller credentials secret by resolving
-	// the reference from the HostedCluster and syncing the secret in the control
-	// plane namespace.
-	var src corev1.Secret
-	if err := c.Get(ctx, client.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.Platform.AWS.KubeCloudControllerCreds.Name}, &src); err != nil {
-		return fmt.Errorf("failed to get cloud controller provider creds %s: %w", hcluster.Spec.Platform.AWS.KubeCloudControllerCreds.Name, err)
-	}
-	dest := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: controlPlaneNamespace,
-			Name:      src.Name,
-		},
-	}
-	_, err := createOrUpdate(ctx, c, dest, func() error {
-		srcData, srcHasData := src.Data["credentials"]
-		if !srcHasData {
-			return fmt.Errorf("hostedcluster cloud controller provider credentials secret %q must have a credentials key", src.Name)
-		}
-		dest.Type = corev1.SecretTypeOpaque
-		if dest.Data == nil {
-			dest.Data = map[string][]byte{}
-		}
-		dest.Data["credentials"] = srcData
-		return nil
-	})
-	if err != nil {
-		return fmt.Errorf("failed to reconcile cloud controller provider creds: %w", err)
-	}
 
-	// Reconcile the platform provider node pool management credentials secret by
-	// resolving  the reference from the HostedCluster and syncing the secret in
-	// the control plane namespace.
-	err = c.Get(ctx, client.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.Platform.AWS.NodePoolManagementCreds.Name}, &src)
-	if err != nil {
-		return fmt.Errorf("failed to get node pool provider creds %s: %w", hcluster.Spec.Platform.AWS.NodePoolManagementCreds.Name, err)
-	}
-	dest = &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: controlPlaneNamespace,
-			Name:      src.Name,
-		},
-	}
-	_, err = createOrUpdate(ctx, c, dest, func() error {
-		srcData, srcHasData := src.Data["credentials"]
-		if !srcHasData {
-			return fmt.Errorf("node pool provider credentials secret %q is missing credentials key", src.Name)
+	awsCredentialsTemplate := `[default]
+role_arn = %s
+web_identity_token_file = /var/run/secrets/openshift/serviceaccount/token
+`
+	// TODO (alberto): consider moving this reconciliation logic down to the CPO.
+	// this is not trivial as the CPO deployment itself needs the secret with the ControlPlaneOperatorARN
+	var errs []error
+	syncSecret := func(secret *corev1.Secret, arn string) error {
+		ns := &corev1.Namespace{}
+		err := c.Get(ctx, client.ObjectKey{Name: secret.Namespace}, ns)
```

Comment:
nit: I don't think there's a need to test the namespace, since all secrets are going to the control plane namespace.

Reply:
dropped!